plugins/woocommerce/includes/gateways/paypal/class-wc-gateway-paypal.php

```diff
@@ -527,27 +527,31 @@ public function should_load() {
 		$should_load = $this->get_option( $option_key );
 
 		if ( '' === $should_load ) {
-
-			// New installs without PayPal Standard enabled don't load it.
-			if ( 'no' === $this->enabled && WC_Install::is_new_install() ) {
-				$should_load = false;
-			} else {
-				$should_load = true;
-			}
+			// Set default `_should_load` to 'yes' on existing stores with PayPal Standard enabled or with existing PayPal Standard orders.
+			$should_load = 'yes' === $this->enabled || $this->has_paypal_orders();
 
 			$this->update_option( $option_key, wc_bool_to_string( $should_load ) );
 		} else {
 			$should_load = wc_string_to_bool( $should_load );
 		}
 
-		/**
-		 * Allow third-parties to filter whether PayPal Standard should be loaded or not.
-		 *
-		 * @since 5.5.0
-		 *
-		 * @param bool              $should_load Whether PayPal Standard should be loaded.
-		 * @param WC_Gateway_Paypal $this        The WC_Gateway_Paypal instance.
-		 */
-		return apply_filters( 'woocommerce_should_load_paypal_standard', $should_load, $this );
```

Comment:
Since this is technically a breaking change I'd rather not simply remove the filter silently. What about this: apply the filter only if we pass `$should_load` as `false`, and if we get it back as `true`, show an admin notice like "We noticed that a plugin or a code snippet in your site uses the woocommerce_should_load_paypal_standard filter to activate PayPal Standard, but we don't support it anymore so it won't be enabled" (this can likely be worded much better, but you get the idea). So:

```php
if(!$should_load) {
  $filtered_should_load = apply_filters( 'woocommerce_should_load_paypal_standard', $should_load, $this );
  if($filtered_should_load) {
    //Add notice
  }
return $should_load;
```

Reply:
+1 to @Konamiman's suggestion to maintain backward compatibility. I suggest we also log whenever this filter is activated. This can be helpful for analytical and debugging purposes in the future whenever this filter is invoked. Something like "Deprecated filter woocommerce_should_load_paypal_standard used in" + trace. In addition, we can also consider pushing it to tracks for monitoring, though I am not entirely sure if this fits the usecase.

 Example:

```php
if($filter_should_load) {
 error_log('Deprecated filter woocommerce_should_load_paypal_standard used in ' . wp_debug_backtrace_summary());
}
```

Reply:
Thanks @Konamiman I've implemented this suggestion in: 022c10579ff59c140460f8a526ed2525666bf87b

Screenshot of the new notice:
![image](https://github.com/woocommerce/woocommerce/assets/2275145/5acda60e-7c55-46e2-8956-b1e2816e952a)

Please let me know if the wording should be updated.

---

@naman03malhotra I tried implementing your suggestion by logging the backtrace inside this condition however it didn't have any information that would help with debugging as it only logs the backtrace of when `WC_Gateway_PayPal->should_load()` is being called, not the code that used the filter.

Apologies if I've misunderstood your suggestion 😅